src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
-    // Redirect the user to the home page
-})
-```
+## Manifest file
 
-### Create a raw auth request
+Blockstack apps have a manifest file based on the [W3C web app manifest specification](https://w3c.github.io/manifest/). The Blockstack Browser retrieves the manifest file from the app during the authentication process and displays some of the information in it such as the app name and icon to the user. The location of the app manifest file is specific in the authentication request token and *MUST* be on the same origin as the app requesting authentication.
 
-```es6
-import { makeAuthRequest, makeECPrivateKey } from 'blockstack'
+Below is an example of a manifest file:
 
-const privateKey = makeECPrivateKey()
-const authRequest = makeAuthRequest(privateKey)
+```
+{
+  "name": "Todo App",
+  "start_url": "http://blockstack-todos.appartisan.com",
+  "description": "A simple todo app build on blockstack",
+  "icons": [{
+    "src": "http://blockstack-todos.appartisan.com/logo.png",
+    "sizes": "400x400",
+    "type": "image/png"
+  }]
+}
 ```
 
-### Verify an auth request
+The manifest file *MUST* have [Cross-origin resource sharing (CORS) headers](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) that allow the manifest file to be fetched from any arbitrary source. This usually means returning  
 
-```es6
-import { verifyAuthRequest } from 'blockstack'
+## Key pairs
 
-const verified = verifyAuthRequest(authRequest)
-```
+Blockstack Authentication makes extensive use of public key cryptography. As mentioned above, we use ECDSA with the secp256k1 curve. What follows is a description of the various public-private key pairs used in the authentication process including how they're generated, where they're used and to whom the private key is disclosed.
+
+### Transit private key
 
-### Create an auth response
+The transit private is an ephemeral key that is used to encrypt secrets that need to be passed from the Blockstack Browser to the app during the authentication process. It is randomly generate by the app at the beginning of the authentication response. The public key that corresponds to the transit private key is stored in a single element array in the `public_keys` key of the authentication request token. The Blockstack Browser encrypts secret data such as the app private using this public key and sends it back to the app when the user signs in to the app. The transit private key signs the app authentication request.
 
-```es6
-import { makeAuthResponse, makeECPrivateKey } from 'blockstack'
+### Blockstack ID Identity address private key
+The identity address private key is derived from the user's keychain phrase and is the private key of the Blockstack ID that the user chooses to use to sign in to the app. It is a secret owned by the user and never leaves the user's instance of the Blockstack browser. This private key signs the authentication response token for an app to indicate that the user approves sign in to that app.
```

Comment:
->
Each Blockstack identity is associated with a private key derived from the user's keychain phrase. The Blockstack ID that signs into an app is derived from a hash of the identity key's corresponding public key (in particular, we use a Base58 check-encoding). This private key is owned by the user and never leaves the user's Blockstack Browser. The identity key signs the authentication response, proving that the user approved the application sign in, however, the identity key _is never_ passed to the application as part of the authentication flow.